simulator/mips/mips_instr.cpp

```diff
@@ -854,19 +852,20 @@ void BaseMIPSInstr<R>::init( const MIPSTableEntry<MyDatapath>& entry, MIPSVersio
 {
     this->imm_print_type = entry.imm_print_type;
     this->set_type( entry.operation);
-    this->mem_size  = entry.mem_size;
-    this->executor  = entry.versions.is_supported(version) ? entry.function : mips_unknown<MyDatapath>;
-    this->dst2      = ( entry.dst == Src::HI_LO) ? MIPSRegister::mips_hi() : MIPSRegister::zero();
-    this->opname    = entry.name;
-    this->print_dst = is_explicit_register( entry.dst);
-    this->print_src1 = !entry.sources.empty()    && is_explicit_register( entry.sources[0]);
-    this->print_src2 =  entry.sources.size() > 1 && is_explicit_register( entry.sources[1]);
-    this->print_src3 =  entry.sources.size() > 2 && is_explicit_register( entry.sources[2]);
+    this->mem_size   = entry.mem_size;
+    this->executor   = entry.versions.is_supported(version) ? entry.function : mips_unknown<MyDatapath>;
+    this->dst.at( 1) = (entry.dst.at( 0) == Src::HI_LO) ? MIPSRegister::mips_hi() : MIPSRegister::zero();
+    this->opname     = entry.name;
+
+    for ( size_t i = 0; i < entry.dst.size(); i++)
+        if ( is_explicit_register( entry.dst.at( i))) this->print_dst.set( i);
```

Comment:
`std::bitset::operator[]` would be more elegant:

```c++
this->print_dst[i] = is_explicit_register( entry.dst[i]);
```

I'll use `std::transform` later